lib/WebService/GoogleAPI/Client.pm

```diff
@@ -316,13 +309,33 @@ sub api_query
         }
         elsif ( $method_discovery_struct->{ parameters }{ $meth_param_spec }{ 'location' } eq 'query' )    ## check post form variables .. assume not get?
         {
+
           if ( !defined $params->{ options }{ $meth_param_spec } )
           {
             $params->{ options }{ $meth_param_spec } = $method_discovery_struct->{ parameters }{ $meth_param_spec }{ default }
               if ( defined $method_discovery_struct->{ parameters }{ $meth_param_spec }{ default } );
           }
 
+	  if ($params->{options}{ $meth_param_spec})
+	  {
+	    #put the query params into the url
+	    if ($params->{path} =~ /\?/) {#check if we've added any query params yet
+	      $params->{path} .= "&$meth_param_spec=$params->{options}{$meth_param_spec}";
+	    } else {
+	      $params->{path} .= "?$meth_param_spec=$params->{options}{$meth_param_spec}";
+	    }
+	  }
+
         }
+
+
+	## undefining it so that it doesn't break post contents
+	#moved undefining here, b/c query params also should be in url
+	delete $params->{ options }{ $meth_param_spec } 
+	    unless $method_discovery_struct->{request}
```

Comment:
this is to allow the same param to be in the payload and the path if needs be

Reply:
There really should not be any cases where a parameter is required both as a GET as well as a FORM or raw JSON BODY - I would certainly be surprised if Google allow this to be acceptable. That said I don't see any harm in loosening the client to allow this. I'm working through the pull - ideally I'd like to allow your approach whilst leaving in my original solution. If you have a usage example of this approach could you message it - I'm just looking to not throw out too much of the validation logic, or at least understand exactly how this works.

Reply:
I think it might be easier for me to just do the merge and then review it - this part of the process is rather new to me.